Replicator/Replicator.cc

```diff
@@ -467,21 +457,14 @@ namespace litecore { namespace repl {
             logVerbose("kWebSocketCloseAppPermanent or kWebSocketCloseAppTransient received, ignoring (only relevant for underlying connection...)");
             return;
         }
-        
+        // Let worker update the error we're holding
         Worker::onError(error);
-        for (const StoppingErrorEntry& stoppingErr : StoppingErrors) {
-            if (stoppingErr.err == error) {
-                string message = error.description().c_str();
-                if (stoppingErr.isFatal) {
-                    logError("Stopping due to fatal error: %s", message.c_str());
-                    _disconnect(websocket::kCloseAppPermanent, stoppingErr.msg);
-                } else {
-                    logError("Stopping due to error: %s", message.c_str());
-                    _disconnect(websocket::kCloseAppTransient, stoppingErr.msg);
-                }
-                return;
-            }
-        }
+
+        // Log error and stop replicator
+        // The logic in c4RemoteReplicator::handleStopped() will decide if we should retry connection
+        string message = error.description();
+        logError("Stopping due to error: %s", message.c_str());
+        _disconnect(websocket::kCloseAppTransient, message);
```

Comment:
Does this mean all errors coming to there will be transient and will cause the replicator to get retired? If this is for fixing hang in busy / error, what is the error that is reported here for that case?

Reply:
The original error Jianmin found that causes Replicator to hang in Busy is if the checkpoints JSON received from remote is unparseable. This is what fuelled the conversation on enforcing the principle of "error always leads to Stopped or Offline". Because we previously have no certainty about the behaviour when encountering an error.
The transient code seems the most suitable to use (to me) out of the available `websocket::CloseCode`s. The actual error that we encountered is still logged, and sent to remote. But also 4001 (kCloseAppTransient) is there with it.